parlai/core/torch_ranker_agent.py

```diff
@@ -167,6 +172,12 @@ def __init__(self, opt: Opt, shared=None):
         opt['rank_candidates'] = True
         super().__init__(opt, shared)
 
+        self.model_parallel = opt.get('model_parallel', False) and self.use_cuda
+        self.data_parallel = opt.get('data_parallel', False) and self.use_cuda
```

Comment:
can we just put this code in TorchAgent instead of in TorchRankerAgent and TorchGeneratorAgent?